src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AdditionalConditionWithElectronicMonitering.kt

```diff
@@ -0,0 +1,186 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AdditionalCondition
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AdditionalConditionData
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.Licence
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+private val dateTimeFormatter = DateTimeFormatter.ofPattern("EEEE dd MMMM yyyy")!!
+private const val CONDITION_CODE_FOR_14B = "524f2fd6-ad53-47dd-8edc-2161d3dd2ed4"
+private const val LED_CHANGE_MSG = "licence end date"
+private const val CRD_OR_ARD_CHANGE_MSG = "release date"
+private const val END_DATE = "endDate"
+
+fun updateAdditionalConditionWithAdditionConditionData(
+  sentenceChanges: SentenceChanges,
+  updatedLicenceEntity: Licence,
+  licenceEntity: Licence
+): AdditionalCondition? {
+  val electronicMonitoringCondition = getElectronicMonitoringCondition(licenceEntity)
+  electronicMonitoringCondition?.let {
+    if (hasCorrectStatusCodeForExistingCondition(licenceEntity)) {
+      calculateEndDateForLedArdCrdChanges(sentenceChanges, updatedLicenceEntity, licenceEntity)?.let { it ->
+        val electronicMonitoringConditionData = electronicMonitoringCondition.additionalConditionData
+          .find { it -> it.dataField == END_DATE }?.copy(
+            dataValue = it
+          )
+        return electronicMonitoringCondition.copy(
+          conditionVersion = updatedLicenceEntity.version!!,
+          additionalConditionData = electronicMonitoringCondition.additionalConditionData.removeExistingEndDateAndAddNew(
+            electronicMonitoringConditionData!!
+          ),
+          expandedConditionText = electronicMonitoringCondition.conditionText?.replace("[INSERT END DATE]", it!!)
+        )
+      }
+    }
+  }
+  return null
+}
+fun getAdditionalConditionDataForCondition(
+  conditionCode: String,
+  licenceEntity: Licence,
+  newAdditionalCondition: AdditionalCondition
+): List<AdditionalConditionData>? {
+  return when {
+    hasAdditionalConditionGotElectronicMonitoringWithCorrectStatusCode(conditionCode, licenceEntity) -> createAdditionalConditionData(licenceEntity, newAdditionalCondition)
+    else -> null
+  }
+}
+fun getReasonForDateChange(
+  hasLedChange: Boolean,
+  updatedLicenceEntity: Licence,
+  licenceEntity: Licence
+): String? {
+  val hasArdOrCrdChanged = hasArdOrCrdChanged(updatedLicenceEntity, licenceEntity)
+  return when {
+    hasLedChange && hasArdOrCrdChanged -> CRD_OR_ARD_CHANGE_MSG.plus(" and ").plus(LED_CHANGE_MSG)
```

Comment:
meh.. string interpolation should generally be used here: 
`"${CRD_OR_ARD_CHANGE_MSG} and $LED_CHANGE_MSG"`